sympy/matrices/matrices.py

```diff
@@ -384,15 +384,56 @@ def cofactor_matrix(self, method="berkowitz"):
                          lambda i, j: self.cofactor(i, j, method))
 
     def det(self, method="bareiss", iszerofunc=None):
-        """Computes the determinant of a matrix.  If the matrix
-        is at most 3x3, a hard-coded formula is used.
-        Otherwise, the determinant using the method `method`.
+        """Computes the determinant of a matrix.
 
+        Parameters
+        ==========
+        method : string, optional
+            Specifies an algorithm to use for computing matrix determinant.
+
+            If the matrix is at most 3x3, the key is ignored and a hard-coded
+            formula is used. Otherwise, it will be defaulted with 'bareiss'.
+
+            If it is set with 'bareiss', Bareiss' fraction-free algorithm would
+            be used
+
+            If it is set with 'berkowitz', Berkowitz algorithm would be used.
+
+            Otherwise, if it is set with 'lu', LU decomposition would be used.
+
+            .. note::
+                For backward compatibility, legacy keys like **bareis** ,
+                **det_lu** can still be used to indicate corresponding
+                methods.
+                And the keys are also **case-insensitive** for now.
+                However, we would discourage you to use the keys in the manner.
+
+        iszerofunc : FunctionType or None, optional
+            Specifies zero testing method used for 'bareiss' or 'lu'.
+            If unspecified, it will be defaulted into None
+
+            If it is set to None, it will be defaulted to _iszero when when
+            method is set to 'bareiss', and _is_zero_after_expand_mul when
+            method is set to 'lu'.
+
+            It can also accept any user-specified zero testing function, if it
+            is formatted as a function which accepts a single symbolic argument
+            and returns True if it is tested as zero and False if it tested as
+            non-zero, and also None if it is undecidable.
+
+        Returns
+        =======
+        det : Basic
+            Result of determinant.
+
+        Raises
+        ======
+        ValueError :
```

Comment:
"ValueError : if unrecognized keys are given for \`method\` or \`iszerofunc\`" (similarly for `NonSquareMatrixError`).